### ISSUE TITLE
Can't slice backed anndata in non-increasing order

### ISSUE THREAD

**Issue by joshua-gould** [theislab/anndata#276]
Generated pbmc3k using scanpy tutorial with no raw data.
```
adata = anndata.read('pbmc3k_no_raw.h5ad', backed='r')
adata[:, ['TNFRSF4', 'SUMO3', 'DSCR3']].X
```

raises TypeError: Indexing elements must be in increasing order

anndata version 0.6.22.post1

**Comment by ivirshup** [comment]
Is this dense or sparse?

This feature seems to be working for me on master – which I think will have a release candidate soon.

**Comment by joshua-gould** [comment]
Dense

**Comment by ivirshup** [comment]
In the current version slicing on dense array still seems to throw an error for negative step size, but indexing in any order should work.

That is, with this setup

```python
# Setup
import scanpy as sc

pbmc = sc.datasets.pbmc3k()
sc.pp.calculate_qc_metrics(pbmc, inplace=True)  # To get total counts for genes
pbmc.write("./tmpdense.h5ad", compression="lzf", as_dense=("X",)) 
backed = sc.read("tmpdense.h5ad", backed="r") 
```

This works:

```python
# First 10 expressed genes in reverse order
indexer = pbmc.var.query("total_counts > 0").index[10::-1]
backed[:, indexer].X
```

This currently doesn't:

```python
backed[:, 10::-1].X
```

**Comment by ivirshup** [comment]
@joshua-gould, I think this is fixed in 0.7, can you confirm?

**Comment by joshua-gould** [comment]
Yes, works now. Thanks.
